### DISCUSSION TITLE
Matomo Wordpress-Plugin

### DISCUSSION THREAD

**Discussion by neralex** [bootscore/bootscore#644] [Q&A] [answered]
Ich versuche das Wordpress-Plugin von Matomo mit diesem Plugin zu verwenden. Den Tracking-Code habe ich in den Plugin-Einstellungen auf "manuell" gesetzt, so dass ich die Attribute des Script-Tags ergänzen kann. Es klappt auch soweit, dass die Attribute bei "Accept" und "Reject" entsprechend entfernt und wieder gesetzt werden. Allerdings werden die beiden erzeugten Cookies _pk_ses.xxxx und _pk_id.xxxx beim Reject nicht gelöscht, obwohl die beiden Cookies vom Script in der Konsole als gefunden und gelöscht markiert werden. Hast du dein Plugin schon einmal mit dem Matomo-Plugin ausprobiert?

```
cookie_table: [           // list of all expected cookies
  {
    col1: '^_pk',
    col2: 'abcdefgh.com',
    col3: '13 months',
    col4: 'Matomo: tracking',
   is_regex: true
  }
```

Beim Reject bekomme ich dann folgendes ausgegeben:
```
CookieConsent [AUTOCLEAR]: search cookie: '^_pk', found: (2) ['_pk_id.1.ac9b', '_pk_ses.1.ac9b']
CookieConsent [AUTOCLEAR]: deleting cookie: '_pk_id.1.ac9b' path: '/' domain: 
(2) ['abcdefgh.com', '.abcdefgh.com']
0: "abcdefgh.com"
1: ".abcdefgh.com"
length: 2
[[Prototype]]: Array(0)
CookieConsent [AUTOCLEAR]: deleting cookie: '_pk_ses.1.ac9b' path: '/' domain: 
(2) ['abcdefgh.com', '.abcdefgh.com']
0: "abcdefgh.com"
1: ".abcdefgh.com"
length: 2
[[Prototype]]: Array(0)
```

Bei bootscore.me steht in der Konsole nach length: 2 zusätzlich noch:
```
lastIndex: 1
lastItem: ".bootscore.me"
```

**Comment by crftwrk** [comment, marked as answer]
Bitte alle Anfragen auf Englisch, ausser uns beiden versteht hier keiner Deutsch ;-)

Yes, I'm using Matomo as well, but not the WP plugin. I use external installations because WP plugin is quiet heavy and external installation allows more control. You can use Matomo without cookies as well, so theoretical you can use it without the Cookie settings plugin. See https://intelligent-heizen.info for an example. 

BTW, bS Cookie Settings does NOT write/delete any external cookies. What it does is to manage and block/allow if tracking script is wrapped in

```html
<script type="text/plain" data-cookiecategory="analytics">
...
</script>
```

tag. 

The bS Cookie Settings is a modification of https://github.com/orestbida/cookieconsent to add Bootstrap and turn script into a WP plugin. No changes made in cookie management. If you have questions about the cookie management, it's better to ask there.

Solved?
